src/api/pagerenderer.cpp

```diff
@@ -0,0 +1,1080 @@
+// File:        pagerenderer.cpp
+// Description: PAGE XML rendering interface
+// Author:      Jan Kamlah
+
+// (C) Copyright 2021
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "errcode.h" // for ASSERT_HOST
+#ifdef _WIN32
+#  include "host.h"  // windows.h for MultiByteToWideChar, ...
+#endif
+#include "tprintf.h" // for tprintf
+
+#include <tesseract/baseapi.h>
+#include <tesseract/renderer.h>
+
+#include <ctime>
+#include <iomanip>
+#include <memory>
+#include <sstream> // for std::stringstream
+#include <regex>
+#include <unordered_set>
+
+#include <allheaders.h>
++#if (LIBLEPT_MAJOR_VERSION == 1 && LIBLEPT_MINOR_VERSION >= 83) || LIBLEPT_MAJOR_VERSION > 1
+#include <array_internal.h>
+#include <pix_internal.h>
+#endif
+
+namespace tesseract {
+
+
+///
+/// Block orientation at the current iterator position.
+///
+static tesseract::Orientation 
+GetBlockTextOrientation(const PageIterator *it) {
```

Comment:
`GetBlockTextOrientation` is unused. There are also currently lots of compiler warnings for unused variables.

Reply:
The file also has several lines with trailing blanks which should be removed.

Reply:
`typos` finds several typos: reveresed, informations, usefull, simplies.